File: MATH/Excel ColumnTitle/ExcelColumTitle.py

/*

Excel Column Title

Given a positive integer, return its corresponding column title as appear in an Excel sheet.

For example:

    1 -> A
    2 -> B
    3 -> C
    ...
    26 -> Z
    27 -> AA
    28 -> AB 

*/


class Solution:
    # @param A : integer
    # @return a strings
    def convertToTitle(self, A):
        res=""
        
        while A>0:
            A-=1
            tmp=A%26;
            A/=26;
            res+=unichr(ord('A')+tmp)
            
        return res[::-1]

File: MATH/Excel ColumnTitle/ExcelColumTitle.java

/*

Excel Column Title

Given a positive integer, return its corresponding column title as appear in an Excel sheet.

For example:

    1 -> A
    2 -> B
    3 -> C
    ...
    26 -> Z
    27 -> AA
    28 -> AB 

*/

public class Solution {
	public String convertToTitle(int a) {
	    
	    StringBuilder res=new StringBuilder();
	    int tmp;
	    
	    while(a>0){
	        a-=1;
	        tmp=a%26;
	        a/=26;
	        res.insert(0,(char)(tmp+'A'));
	    }
	    return res.toString();
	}
}

